### PR TITLE
Fr us 25 refactor

### LINKED ISSUE #126: Must be user to checkout US 25
As a visitor. they should not see the "checkout" link in the cart until they have registered.
The message "must either login or register" is displaying correctly but we need to hide the "checkout" button in the cart view page.

### PR DESCRIPTION
**Pull request by fredrondina**
added requirement that user be logged in, in order to checkout

Edited checkout tests so that users are logged in, per new requirement 

closes #126

### CODE REVIEW

**Review by kelshadarby — APPROVED** [review]
Nice work on this Fred!

File: app/views/cart/show.html.erb [hunk 1]
```diff
@@ -49,7 +49,7 @@
     <% if current_user == nil%>
       <p>You must <%= link_to "log in", "/login"%> or <%= link_to "register", "/register"%> to checkout</p>
     <%end%>
-  <p><%= link_to "Checkout", "/orders/new", method: :get %></p>
+  <p><%= link_to "Checkout", "/orders/new", method: :get if current_user%></p>
```
Comment (brian-greeson) [inline app/views/cart/show.html.erb:52]: Perfect

File: spec/features/cart/checkout_spec.rb [hunk 2]
```diff
@@ -16,9 +16,11 @@
       visit "/items/#{@pencil.id}"
       click_on "Add To Cart"
       @items_in_cart = [@paper,@tire,@pencil]
+      @user = User.create!( email_address: 'user1@example.com', password: 'password', role: 'default', name: 'User 1', street_address: '123 Example St', city: 'Userville', state: 'State 1', zip_code: '12345')
     end
 
-    it 'Theres a link to checkout' do
+    it 'Theres a link to checkout when logged in' do
+      allow_any_instance_of(ApplicationController).to receive(:current_user).and_return(@user)
```
Comment (brian-greeson) [inline spec/features/cart/checkout_spec.rb:23]: Great Mock

File: spec/features/cart/checkout_spec.rb [hunk 3]
```diff
@@ -28,11 +30,12 @@
       expect(current_path).to eq("/orders/new")
     end
 
-    it 'There is not a link to checkout, only to login or register' do
+    it 'There is not a link to checkout, only to login or register when visiting' do
```
Comment (brian-greeson) [inline spec/features/cart/checkout_spec.rb:33]: Nice clarification

**Review by brian-greeson — APPROVED** [review]
Looks great

**Pull request merged**